graphql_compiler/compiler/blocks.py

```diff
@@ -207,6 +207,10 @@ def to_gremlin(self):
         mark_name, _ = self.location.get_location_name()
         return u'as({})'.format(safe_quoted_string(mark_name))
 
+    def __hash__(self):
```

Comment:
It's a bit weird to have this one specific subclass of BasicBlock be hashable, and nothing else. Let's discuss this in person and see if we can avoid this.